test/topo/topotest.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-config/test/utils/gnmi"
+	toposdk "github.com/onosproject/onos-ric-sdk-go/pkg/topo"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// NewTopoClient creates a new topo SDK client
+func NewTopoClient() (toposdk.Client, error) {
+	return toposdk.NewClient()
+}
+
+func checkTopo(t *testing.T) {
+	const (
+		onosConfigKind = topo.ONOS_CONFIG
+		devicesimKind  = "Devicesim"
+	)
+	var (
+		configEntity *topo.Entity
+		configObject topo.Object
+		targetEntity *topo.Entity
+		targetObject topo.Object
+
+		configID topo.ID
+		targetID topo.ID
+
+		relation       *topo.Relation
+		relationObject topo.Object
+	)
+
+	// Get a topology API client
+	client, err := NewTopoClient()
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	// Get the list of topo objects
+	objs, err := client.List(context.Background())
```

Comment:
You can use filters to get entities or relations rather than listing all of the objects.

Reply:
like this https://github.com/onosproject/onos-e2t/blob/master/test/utils/topo.go#L275